File: server/app/controller/subscription_controller.py
from fastapi import APIRouter, Response, Request, HTTPException
from app.schema.subscription_schema import CreateSubscription
from app.service.subscription_service import SubscriptionService

subscription_router = APIRouter()


@subscription_router.post("/", status_code=201)
async def create_subscription(subscription: CreateSubscription):
    new_subscription = SubscriptionService.create(payload=subscription)
    if new_subscription:
        return {"message": "Subscription has been created."}
    else:
        raise HTTPException(status_code=400, detail="User not created!")

File: node-server/src/controllers/subscription-controller.ts
import { Exception } from '../exceptions/exception';
import { SubscriptionService } from '../services/subscription-service';
import { Response, Request } from 'express';



export class SubscriptionController {
    private subscriptionService: SubscriptionService;

    constructor() {
        this.subscriptionService = new SubscriptionService();
    }

    public create = async (req: Request, res: Response) => {
        try {
            const userId = req.user.id;
            const payload = req.body;
            await this.subscriptionService.create(payload, userId);
            res.status(201).json({ message: 'Subscription created.' });
        } catch (error) {
            if (error instanceof Exception) {
                res.status(error.statusCode).json({ message: error.message });
            } else {
                res.status(500).json({ message: 'Internal server error!' });
            }
        }
    };

    public delete = async (req: Request, res: Response) => { };
}
